### ISSUE TITLE
[BUG] Spring 4 documentation: provisioning for multibinder scenario

### ISSUE THREAD

**Issue by lmolkova** [Azure/azure-sdk-for-java#27058]
I've tried to configure [provisioning of servicebus queues](https://github.com/Azure-Samples/azure-spring-boot-samples/blob/spring-cloud-azure_4.0.0-beta.4/servicebus/spring-cloud-azure-stream-binder-servicebus/servicebus-queue-binder/src/main/resources/application-rm.yaml) along with [multibinder sample](https://github.com/Azure-Samples/azure-spring-boot-samples/tree/spring-cloud-azure_4.0.0-beta.4/servicebus/spring-cloud-azure-stream-binder-servicebus/servicebus-multibinders)

I've tried putting profile and resource config in corresponding binder sections or setting a global one, but none of these approaches work.

It should be documented how it can be done (with a sample or in other ways) or limitations (if it cannot be done).

Here's my config:

```yml
spring:
  cloud:
    azure:
      credential:
        client-id: client
        client-secret: secret
        tenant-id: tenant
      profile:
        subscription-id: id
      servicebus:
        resource:
          resource-group: group

    stream:
      function:
        definition: consume1;supply1;consume2;supply2
      bindings:
        consume1-in-0:
          binder: servicebus-1
          destination: testnew1
        supply1-out-0:
          binder: servicebus-1
          destination: testnew1
        consume2-in-0:
          binder: servicebus-2
          destination: testnew2
        supply2-out-0:
          binder: servicebus-2
          destination: testnew2
      binders:
        servicebus-1:
          type: servicebus
          default-candidate: true
          environment:
            spring:
              cloud:
                azure:
                  servicebus:
                    namespace: startertest2

        servicebus-2:
          type: servicebus
          default-candidate: false
          environment:
            spring:
              cloud:
                azure:
                  servicebus:
                    namespace: startertest2

      servicebus:
        bindings:
          consume1-in-0:
            consumer:
              checkpoint-mode: MANUAL
          supply1-out-0:
            producer:
              entity-type: queue
          consume2-in-0:
            consumer:
              checkpoint-mode: MANUAL
          supply2-out-0:
            producer:
              entity-type: queue
      poller:
        fixed-delay: 1000
        initial-delay: 0
```

or

```yml
spring:
  cloud:
    stream:
      function:
        definition: consume1;supply1;consume2;supply2
      bindings:
        consume1-in-0:
          binder: servicebus-1
          destination: testnew1
        supply1-out-0:
          binder: servicebus-1
          destination: testnew1
        consume2-in-0:
          binder: servicebus-2
          destination: testnew2
        supply2-out-0:
          binder: servicebus-2
          destination: testnew2
      binders:
        servicebus-1:
          type: servicebus
          default-candidate: true
          environment:
            spring:
              cloud:
                azure:
                  credential:
                    client-id: client
                    client-secret: secret
                    tenant-id: tenant
                  profile:
                    subscription-id: id

                  servicebus:
                    namespace: startertest2
                    resource:
                      resource-group: group


        servicebus-2:
          type: servicebus
          default-candidate: false
          environment:
            spring:
              cloud:
                azure:
                  credential:
                    client-id: client
                    client-secret: secret
                    tenant-id: tenant
                  profile:
                    subscription-id: id

                  servicebus:
                    namespace: startertest2
                    resource:
                      resource-group: group


      servicebus:
        bindings:
          consume1-in-0:
            consumer:
              checkpoint-mode: MANUAL
          supply1-out-0:
            producer:
              entity-type: queue
          consume2-in-0:
            consumer:
              checkpoint-mode: MANUAL
          supply2-out-0:
            producer:
              entity-type: queue
      poller:
        fixed-delay: 1000
        initial-delay: 0
```

error:

```
rror starting ApplicationContext. To display the conditions report re-run your application with 'debug' enabled.
2022-02-11 13:40:21.839 ERROR 14056 --- [           main] o.s.boot.SpringApplication               : Application run failed

org.springframework.context.ApplicationContextException: Failed to start bean 'outputBindingLifecycle'; nested exception is org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'azureResourceManager' defined in com.azure.spring.cloud.autoconfigure.resourcemanager.AzureResourceManagerAutoConfiguration: Unsatisfied dependency expressed through method 'azureResourceManager' parameter 0; nested exception is org.springframework.beans.factory.NoSuchBeanDefinitionException: No qualifying bean of type 'com.azure.core.credential.TokenCredential' available: expected at least 1 bean which qualifies as autowire candidate. Dependency annotations: {}
	at org.springframework.context.support.DefaultLifecycleProcessor.doStart(DefaultLifecycleProcessor.java:181) ~[spring-context-5.3.14.jar:5.3.14]
	at org.springframework.context.support.DefaultLifecycleProcessor.access$200(DefaultLifecycleProcessor.java:54) ~[spring-context-5.3.14.jar:5.3.14]
	at org.springframework.context.support.DefaultLifecycleProcessor$LifecycleGroup.start(DefaultLifecycleProcessor.java:356) ~[spring-context-5.3.14.jar:5.3.14]
	at java.base/java.lang.Iterable.forEach(Iterable.java:75) ~[na:na]
	at org.springframework.context.support.DefaultLifecycleProcessor.startBeans(DefaultLifecycleProcessor.java:155) ~[spring-context-5.3.14.jar:5.3.14]
	at org.springframework.context.support.DefaultLifecycleProcessor.onRefresh(DefaultLifecycleProcessor.java:123) ~[spring-context-5.3.14.jar:5.3.14]
	at org.springframework.context.support.AbstractApplicationContext.finishRefresh(AbstractApplicationContext.java:935) ~[spring-context-5.3.14.jar:5.3.14]
	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:586) ~[spring-context-5.3.14.jar:5.3.14]
	at org.springframework.boot.web.servlet.context.ServletWebServerApplicationContext.refresh(ServletWebServerApplicationContext.java:145) ~[spring-boot-2.6.2.jar:2.6.2]
	at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:730) ~[spring-boot-2.6.2.jar:2.6.2]
	at org.springframework.boot.SpringApplication.refreshContext(SpringApplication.java:412) ~[spring-boot-2.6.2.jar:2.6.2]
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:302) ~[spring-boot-2.6.2.jar:2.6.2]
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1301) ~[spring-boot-2.6.2.jar:2.6.2]
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1290) ~[spring-boot-2.6.2.jar:2.6.2]
	at com.azure.spring.sample.servicebus.queue.multibinders.ServiceBusQueueMultiBindersApplication.main(ServiceBusQueueMultiBindersApplication.java:27) ~[classes/:na]
Caused by: org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'azureResourceManager' defined in com.azure.spring.cloud.autoconfigure.resourcemanager.AzureResourceManagerAutoConfiguration: Unsatisfied dependency expressed through method 'azureResourceManager' parameter 0; nested exception is org.springframework.beans.factory.NoSuchBeanDefinitionException: No qualifying bean of type 'com.azure.core.credential.TokenCredential' available: expected at least 1 bean which qualifies as autowire candidate. Dependency annotations: {}
	at org.springframework.beans.factory.support.ConstructorResolver.createArgumentArray(ConstructorResolver.java:800) ~[spring-beans-5.3.14.jar:5.3.14]
	at org.springframework.beans.factory.support.ConstructorResolver.instantiateUsingFactoryMethod(ConstructorResolver.java:541) ~[spring-beans-5.3.14.jar:5.3.14]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.instantiateUsingFactoryMethod(AbstractAutowireCapableBeanFactory.java:1352) ~[spring-beans-5.3.14.jar:5.3.14]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBeanInstance(AbstractAutowireCapableBeanFactory.java:1195) ~[spring-beans-5.3.14.jar:5.3.14]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:582) ~[spring-beans-5.3.14.jar:5.3.14]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:542) ~[spring-beans-5.3.14.jar:5.3.14]
	at org.springframework.beans.factory.support.AbstractBeanFactory.lambda$doGetBean$0(AbstractBeanFactory.java:335) ~[spring-beans-5.3.14.jar:5.3.14]
	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:234) ~[spring-beans-5.3.14.jar:5.3.14]
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:333) ~[spring-beans-5.3.14.jar:5.3.14]
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:208) ~[spring-beans-5.3.14.jar:5.3.14]
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.preInstantiateSingletons(DefaultListableBeanFactory.java:953) ~[spring-beans-5.3.14.jar:5.3.14]
	at org.springframework.context.support.AbstractApplicationContext.finishBeanFactoryInitialization(AbstractApplicationContext.java:918) ~[spring-context-5.3.14.jar:5.3.14]
	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:583) ~[spring-context-5.3.14.jar:5.3.14]
	at org.springframework.cloud.stream.binder.DefaultBinderFactory.initializeBinderContextSimple(DefaultBinderFactory.java:471) ~[spring-cloud-stream-3.2.1.jar:3.2.1]
	at org.springframework.cloud.stream.binder.DefaultBinderFactory.getBinderInstance(DefaultBinderFactory.java:265) ~[spring-cloud-stream-3.2.1.jar:3.2.1]
	at org.springframework.cloud.stream.binder.DefaultBinderFactory.doGetBinder(DefaultBinderFactory.java:223) ~[spring-cloud-stream-3.2.1.jar:3.2.1]
	at org.springframework.cloud.stream.binder.DefaultBinderFactory.getBinder(DefaultBinderFactory.java:151) ~[spring-cloud-stream-3.2.1.jar:3.2.1]
	at org.springframework.cloud.stream.binding.BindingService.getBinder(BindingService.java:389) ~[spring-cloud-stream-3.2.1.jar:3.2.1]
	at org.springframework.cloud.stream.binding.BindingService.bindProducer(BindingService.java:273) ~[spring-cloud-stream-3.2.1.jar:3.2.1]
	at org.springframework.cloud.stream.binding.BindingService.bindProducer(BindingService.java:297) ~[spring-cloud-stream-3.2.1.jar:3.2.1]
	at org.springframework.cloud.stream.binding.BindingService.bindProducer(BindingService.java:301) ~[spring-cloud-stream-3.2.1.jar:3.2.1]
	at org.springframework.cloud.stream.binding.AbstractBindableProxyFactory.createAndBindOutputs(AbstractBindableProxyFactory.java:142) ~[spring-cloud-stream-3.2.1.jar:3.2.1]
	at org.springframework.cloud.stream.binding.OutputBindingLifecycle.doStartWithBindable(OutputBindingLifecycle.java:58) ~[spring-cloud-stream-3.2.1.jar:3.2.1]
	at java.base/java.util.LinkedHashMap$LinkedValues.forEach(LinkedHashMap.java:647) ~[na:na]
	at org.springframework.cloud.stream.binding.AbstractBindingLifecycle.start(AbstractBindingLifecycle.java:57) ~[spring-cloud-stream-3.2.1.jar:3.2.1]
	at org.springframework.cloud.stream.binding.OutputBindingLifecycle.start(OutputBindingLifecycle.java:34) ~[spring-cloud-stream-3.2.1.jar:3.2.1]
	at org.springframework.context.support.DefaultLifecycleProcessor.doStart(DefaultLifecycleProcessor.java:178) ~[spring-context-5.3.14.jar:5.3.14]
	... 14 common frames omitted
Caused by: org.springframework.beans.factory.NoSuchBeanDefinitionException: No qualifying bean of type 'com.azure.core.credential.TokenCredential' available: expected at least 1 bean which qualifies as autowire candidate. Dependency annotations: {}
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.raiseNoMatchingBeanFound(DefaultListableBeanFactory.java:1799) ~[spring-beans-5.3.14.jar:5.3.14]
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.doResolveDependency(DefaultListableBeanFactory.java:1355) ~[spring-beans-5.3.14.jar:5.3.14]
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.resolveDependency(DefaultListableBeanFactory.java:1309) ~[spring-beans-5.3.14.jar:5.3.14]
	at org.springframework.beans.factory.support.ConstructorResolver.resolveAutowiredArgument(ConstructorResolver.java:887) ~[spring-beans-5.3.14.jar:5.3.14]
	at org.springframework.beans.factory.support.ConstructorResolver.createArgumentArray(ConstructorResolver.java:791) ~[spring-beans-5.3.14.jar:5.3.14]
	... 40 common frames omitted
```

**Comment by yiliuTo** [comment]
@lmolkova  thanks for reporting the bug, and this has been fixed in pr #26792 which will be released in the version 4.0.0-beta.4. 

Besides, the prefix of property of `tenant-id` should be `spring.cloud.azure.profile.` instead of `spring.cloud.azure.credential.` And for provisioning resouces, anther property of `spring.cloud.stream.servicebus.bindings.<binding-name>.consumer.entity-type` need also be configured, we will also mention this point in our docs. So after the bug fix and correct configuration, the resource provisioning of multibinders can work as expected.

**Comment by lmolkova** [comment]
nice, it works with fresh branch, thanks!
